### PR TITLE
CFWEB-177 Update Styleguide app to Angular 18

### PR DESCRIPTION
**Pull request by ssaring**
Includes:
- update to Angular 18.1
- update to TypeScript 5.4
- update to new Angular build system (see https://angular.dev/tools/cli/build-system-migration)
- update to @angular-eslint/schematics@v18
- update to NGRX 18

### CODE REVIEW

**Comment by MariusAquarius** [comment]
The triggered build does fail on this pull request, this happens because of outdated configurations for wdio, which will be fixed in [this PR](https://github.com/ZEISS/angular-styleguide/pull/26).

File: src/app/catalog/recommendation/services/recommendation.service.spec.ts [hunk 1]
```diff
@@ -17,8 +18,12 @@ describe('RecommendationService', () => {
 
   beforeEach(() => {
     TestBed.configureTestingModule({
-      providers: [RecommendationService],
-      imports: [HttpClientTestingModule],
+      imports: [],
+      providers: [
+        RecommendationService,
+        provideHttpClient(withInterceptorsFromDi()),
```
Comment (Azaraky) [inline src/app/catalog/recommendation/services/recommendation.service.spec.ts:24]: I found this on the official Angular page for "withInterceptorsFromDi".
https://angular.dev/api/common/http/withInterceptorsFromDi?tab=description
Prefer [withInterceptors](https://angular.dev/api/common/http/withInterceptors) and functional interceptors instead, as support for DI-provided interceptors may be phased out in a later release.

Maybe you should check this.

**Review by Azaraky — APPROVED** [review]
Added a comment

File: src/app/catalog/recommendation/services/recommendation.service.spec.ts (hunk 1 above)
Comment (ssaring) [reply src/app/catalog/recommendation/services/recommendation.service.spec.ts:24]: Good hint, thanks. Turns out that the withInterceptor...() can be removed completely in all places.

**Pull request merged**